packages/theme/src/cli/services/dev.ts

```diff
@@ -49,6 +52,10 @@ export async function dev(options: DevOptions) {
     return
   }
 
+  const storefrontPassword = (await isStorefrontPasswordProtected(options.adminSession.storeFqdn))
+    ? await ensureValidPassword(options.storePassword, options.adminSession.storeFqdn)
+    : undefined
```

Comment:
In this case, I believe it's important to adopt the synchronous approach because we need input from users. I've recorded a video (sent you on Slack) with a demo comparing how the new and old workflows handle password protection.

Please feel free to drop a ping if I misunderstood something, and I can apply an update in a subsequent PR. Thank you for the review! :D